src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -212,6 +220,39 @@ instance Buildable TxOut where
 instance Buildable (TxIn, TxOut) where
     build (txin, txout) = build txin <> " ==> " <> build txout
 
+data TxMeta = TxMeta
+    { metaId :: !(Hash "Tx")
+    -- FIXME: we might want to go for something more optimized than natural,
+    -- like Word64 ?
+    , depth :: !(Quantity "block" Natural)
```

Comment:
do we want to keep types in `Primitives` in their optimized form , thus opting for a bit more performant type?

Reply:
Optimizations between `Natural` vs `Word64` is pointless at this stage. Writing an algorithm that goes in O(n²) will cause us some trouble. Using `Natural` instead of `Word64` won't probably have any noticeable effects on a wallet. 

If so, we should _measure_ it first and have a baseline to compare with.

Reply:
I will leave it to `Natural` then because its a safer type. If there will be some evidence that this should be optimized more we can then measure and optimize